File: remove_duplicates_from_linked_list/remove_duplicates_from_linked_list.java
class LinkedList {
    int value;
    LinkedList next;

    LinkedList(int value) {
        this.value = value;
        this.next = null;
    }

    public static LinkedList removeDuplicatesFromLinkedList(LinkedList linkedList) {
        LinkedList currentNode = linkedList;
        while (currentNode != null) {
            LinkedList nextNode = currentNode.next;
            if (nextNode != null && currentNode.value == nextNode.value) {
                currentNode.next = nextNode.next;
            } else {
                currentNode = currentNode.next;
            }
        }
        return linkedList;
    }

    public static void main(String[] args) {
        LinkedList linkedList = new LinkedList(1);
        linkedList.next = new LinkedList(1);
        linkedList.next.next = new LinkedList(2);

        linkedList = removeDuplicatesFromLinkedList(linkedList);

        System.out.println(linkedList.value);
        System.out.println(linkedList.next.value);
    }
}

File: remove_duplicates_from_linked_list/remove_duplicates_from_linked_list.py
from os import link

def remove_duplicates_from_linked_list(linked_list):
  current_node = linked_list
  while current_node is not None:
    next_node = current_node.next
    if next_node is not None:
      if current_node.value == next_node.value:
        current_node.value = next_node.value
        current_node.next = next_node.next
      else:
        current_node = current_node.next
    else:
      current_node = current_node.next

    return linked_list

class LinkedList:
  def __init__(self,value) -> None:
      self.value = value
      self.next = None

linked_list = LinkedList(1)
linked_list.next = LinkedList(1)
linked_list.next.next = LinkedList(2)
linked_list = remove_duplicates_from_linked_list(linked_list)

print(
  linked_list.value,
  linked_list.next.value
)